cluster/proxy_service.py

```
@@ -0,0 +1,267 @@
+        # - Insert/Update: {"h": [hashes]}
+        # - Delete: {}
+        if operation == "delete":
+            message = {}
+        else:
+            message = {"h": data["h"]} if "h" in data else {}
+        
+        try:
+            await self.js.publish(subject, json.dumps(message).encode())
+            logger.debug(f"Published to {subject}: {operation} -> {message}")
+        except Exception as e:
+            logger.error(f"Failed to publish to NATS: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)